src/validators/feed_info.rs

```diff
@@ -0,0 +1,119 @@
+use crate::validators::issues::*;
+
+pub fn validate(gtfs: &gtfs_structures::Gtfs) -> Vec<Issue> {
+    let missing_url = gtfs
+        .feed_info
+        .iter()
+        .filter(|feed_info| !has_url(feed_info))
+        .map(|feed_info| make_issue(feed_info, IssueType::MissingUrl));
+    let invalid_url = gtfs
+        .feed_info
+        .iter()
+        .filter(|feed_info| !valid_url(feed_info))
+        .map(|feed_info| {
+            make_issue(feed_info, IssueType::InvalidUrl)
+                .details(&format!("Publisher url {} is invalid", feed_info.url))
+        });
+    let missing_lang = gtfs
+        .feed_info
+        .iter()
+        .filter(|feed_info| !has_lang(feed_info))
+        .map(|feed_info| make_issue(feed_info, IssueType::MissingLanguage));
+    let invalid_lang = gtfs
+        .feed_info
+        .iter()
+        .filter(|feed_info| !valid_lang(feed_info))
+        .map(|feed_info| {
+            make_issue(feed_info, IssueType::InvalidLanguage)
+                .details(&format!("Language code {} does not exist", feed_info.lang))
+        });
+    missing_url
+        .chain(invalid_url)
+        .chain(missing_lang)
+        .chain(invalid_lang)
+        .collect()
+}
+
+fn make_issue(feed: &gtfs_structures::FeedInfo, issue_type: IssueType) -> Issue {
+    Issue::new(Severity::Error, issue_type, "").name(&format!("{}", feed))
+}
+
+fn has_url(feed: &gtfs_structures::FeedInfo) -> bool {
+    !feed.url.is_empty()
+}
+
+fn valid_url(feed: &gtfs_structures::FeedInfo) -> bool {
```

Comment:
minor: you can rewrite this as:
`url::Url::parse(feed.url.as_ref()).map_or(|_| false, |url| vec!["https", "http", "ftp"].contains(&url.scheme()))`

alternativly (I'm not sure it's better though) you can do:

```rust
url::Url::parse(feed.url.as_ref()).map_or(|_| false, |url| match url.scheme.as_ref() {
"https" | "http" | "ftp" => true,
_ => false,
```